lib/src/bazelify/initialize.dart

```diff
@@ -186,80 +187,167 @@ class _GitTagRulesSource implements DartRulesSource {
       ')\n';
 }
 
-/// Runs `bazelify init` as specified in [arguments].
-Future<Null> initalize(BazelifyInitArguments arguments) async {
-  // Start timing.
-  final timings = <String, Duration>{};
-  final stopwatch = new Stopwatch()..start();
-
-  // Store and change the CWD.
-  var previousCurrent = Directory.current;
-  Directory.current = new Directory(arguments.pubPackageDir);
-
-  // Run "pub get".
-  await Process.run(arguments.pubExecutable, const ['get']);
-  timings['pub get'] = stopwatch.elapsed;
-  stopwatch.reset();
-
-  // Revert back to the old CWD
-  Directory.current = previousCurrent;
-
-  // Read the package's pubspec and the generated .packages file.
-  final pubspec = await Pubspec.fromPackageDir(arguments.pubPackageDir);
-  final packagesFilePath = p.join(arguments.pubPackageDir, '.packages');
-  final packages = parse(
-    await new File(packagesFilePath).readAsBytes(),
-    Uri.parse(packagesFilePath),
-  );
-
-  // Clean and re-build the .bazelify folder.
-  final bazelifyPath = p.join(arguments.pubPackageDir, '.bazelify');
-  final bazelifyDir = new Directory(bazelifyPath);
-  if (await bazelifyDir.exists()) {
-    await bazelifyDir.delete(recursive: true);
+class _Initialize {
+  final BazelifyInitArguments arguments;
+
+  _Initialize(this.arguments);
+
+  Future<Null> run() async {
+    final timings = <String, Duration>{};
+    final stopwatch = new Stopwatch()..start();
+    await _pubGetInPackage();
```

Comment:
might want to create a function to do this, something like:

```dart
// Ideally not async as it won't start the timer right away.... maybe doesn't matter
Future<Duration> _timeAsyncOperation(Future<Null> operation()) {
  var stopwatch = new Stopwatch()..start();
  return operation().then((_) {
    stopwatch.stop();
    return stopwatch.elapsed;
  });
}
```

This would clean things up a bit and make sure we properly do the stopwatch management all the time.